### DISCUSSION TITLE
Airsim+Unity: tornado.general: Connect error   WSAECONNREFUSED

### DISCUSSION THREAD

**Discussion by gwnudt** [microsoft/AirSim#4132] [Support Q&A]
I installed  airsim + unity 2019.3.12 as https://microsoft.github.io/AirSim/Unity/ and
run the hello_car.py   but encountered the error
"WARNING:tornado.general:Connect error on fd 508: WSAECONNREFUSED" 
and can not start the drone scene with the problem ""Problem in starting AirSim server!!!""

I then tried unity 2019.2.11 and the two scenes could start but 
when ran hello_car.py, the connection error still exist "WARNING:tornado.general:Connect error on fd 508: WSAECONNREFUSED" 
but can succeedfully run hello_drone.py 

in addition, it works when I run airsim with unreal. 

I can not figure out the problem and solve it for a long time. Can anyone please help? thanks  a lot

**Comment by zimmy87** [comment]
Is Unity configured to be allowed through your firewall? If you're on windows, you can check whether the Unity Editor is allowed through Windows Firewall by going to Control Panel\System and Security\Windows Defender Firewall\Allowed apps and making sure Unity Editor is checked.
